airflow/providers/amazon/aws/hooks/lambda_function.py

```diff
@@ -40,30 +41,84 @@ class LambdaHook(AwsBaseHook):
 
     def __init__(
         self,
-        function_name: str,
-        log_type: str = 'None',
-        qualifier: str = '$LATEST',
-        invocation_type: str = 'RequestResponse',
         *args,
         **kwargs,
     ) -> None:
-        self.function_name = function_name
-        self.log_type = log_type
-        self.invocation_type = invocation_type
-        self.qualifier = qualifier
         kwargs["client_type"] = "lambda"
         super().__init__(*args, **kwargs)
 
-    def invoke_lambda(self, payload: str) -> str:
-        """Invoke Lambda Function"""
-        response = self.conn.invoke(
-            FunctionName=self.function_name,
-            InvocationType=self.invocation_type,
-            LogType=self.log_type,
-            Payload=payload,
-            Qualifier=self.qualifier,
-        )
+    def invoke_lambda(
+        self,
+        function_name: str,
+        invocation_type: Optional[str] = None,
+        log_type: Optional[str] = None,
+        client_context: Optional[str] = None,
+        payload: Optional[str] = None,
+        qualifier: Optional[str] = None,
+    ):
```

Comment:
```suggestion
    def invoke_lambda(
        self,
        *,
        function_name: str,
        invocation_type: Optional[str] = None,
        log_type: Optional[str] = None,
        client_context: Optional[str] = None,
        payload: Optional[str] = None,
        qualifier: Optional[str] = None,
    ):
```

How about make these all keyword-only arguments? This is probably how a user is supposed to use them anyway (the invocation line would not be readable otherwise), so let’s enforce them.

Same for other functions on the hook.

Reply:
Hello @uranusjr, these were originally kwargs. However, based on @malthe's feedback (https://github.com/apache/airflow/pull/21686#discussion_r810589155) from an earlier discussion (which I agree with), it may be difficult for first timers to understand what is required and the typing info helps as well. 
As we are building this for Airflow, it makes sense to spell out these parameters for users for better reference. 

What are your thoughts? If the above does not make sense, I'll go ahead and make the required changes.

Reply:
I do not suggest changing this to kwargs, but _keyword-only arguments_.

https://www.python.org/dev/peps/pep-3102/

Reply:
@schirag1993 yeah it's just adding a magic "*" argument to denote "you shall not pass" positional parameters after this.

Reply:
Ah I misunderstood then. Thank you for the clarifications @uranusjr and @malthe! 
Will incorporate the changes.

Reply:
Done @uranusjr! Thank you for the feedback. Today I learnt something new